### ISSUE TITLE
image 4.8-0 does not seem to contain correct version information

### ISSUE THREAD

**Issue by killarny** [zulip/docker-zulip#330]
After upgrading to `zulip/docker-zulip:4.8-0` from `zulip/docker-zulip:4.7-0`, I noticed that the version information in the about dialog seems wrong:

![image](https://user-images.githubusercontent.com/29001/144923045-fbf613ee-a2cf-493f-979f-c46baa970890.png)

I'm fairly sure that in `zulip/docker-zulip:4.7-0` this dialog showed `4.7-0` as expected, and this new version output is confusing.

Here's my command output verifying that I'm using the image I think I am:

```
$ docker-compose images
      Container                Repository          Tag       Image Id       Size
----------------------------------------------------------------------------------
chat-zulip_database_1    zulip/zulip-postgresql   10       5e46b560b5be   107.4 MB
chat-zulip_memcached_1   memcached                alpine   4fd2f3aa520a   8.224 MB
chat-zulip_rabbitmq_1    rabbitmq                 3.7.7    d502f687f36a   124.8 MB
chat-zulip_redis_1       redis                    alpine   3900abf41552   32.38 MB
chat-zulip_zulip_1       zulip/docker-zulip       4.8-0    040b8dd651c7   1.955 GB
```

Additionally, it appears that even the cli doesn't report the version correctly (just blank strings, apparently):

```
$ docker-compose exec zulip /sbin/entrypoint.sh app:version
This container contains:
> Zulip server
> Checksum:
```

**Comment by timabbott** [comment]
Thanks for the report!  I suspect the image is being built from a shallow clone of the Zulip git repository, and thus doesn't have the metadata needed to calculate this.

@alexmv FYI

**Comment by timabbott** [comment]
I guess I'll also mention that the value is still uniquely mappable to the version.  Based on those values, I think you're running `upstream/main` aka 5.0, not 4.8, and have a few commits on your own fork?

Given those details, I suspect the problem is that you haven't done a `git fetch --tags` in whatever repository you're building the image from since Zulip 2.2 was released.

```
tabbott@coset:~/zulip$ git describe --always --tags --match='2.[0-9]*' -- 4.8 
2.2-dev-8642-gae000bfdba
tabbott@coset:~/zulip$ git describe --always --tags --match='2.[0-9]*' -- upstream/main 
2.2-dev-11477-gf533097487
```

**Comment by alexmv** [comment]
The 4.8-0 build is actually 0a2649d9fe2b363c668b671c77fbab2fd0a554b3 plus 358a7fb0c6 and f31bf3f06c, due to some form of bad caching during build.

The good news is that means that `camo` works in the image, which wasn't expected to be true until 4.9.

The _very bad_ news is that means that you, and everyone else running "`4.8-0`" are actually running `main`, and thus your upgrade path means you must continue to upgrade through `main`, at least until 5.0 is released.

I've pulled the 4.8-0 image until we can decide what to do.  We can release a 4.8-1 which is actually 4.8, but that will fail to upgrade from 4.8-0.

**Comment by killarny** [comment]
Oh, nice I did notice that YT previews are working again for the first time in ages, so that's awesome! 😅 (That's what camo does, right?)

> your upgrade path means you must continue to upgrade through main, at least until 5.0

Could you elaborate on that process a bit? I don't mind the extra steps, just want to be sure I know what they are.

**Comment by alexmv** [comment]
I've released 4.8-1, which is actually 4.8.

To upgrade along `main`, you can either:
 - Do nothing now, and upgrade to the 5.0-beta1 release which we intend to push out next week, or the 5.0 final release when it comes out (likely in January).
 - Clone this repo, and adjust the `ZULIP_GIT_REF` in `docker-compose.yml` to say `main`, which will build from the latest commit on `main`.
 
 The key is that you cannot upgrade to 4.8-1, as it's not actually an upgrade.

**Comment by killarny** [comment]
Sounds great, thank you for the details and the new image, @alexmv! That's not so bad after all. As far as I can tell in our case, what I'm running is working well for us, so I'll just wait for 5.0 to update again!
